src/main/java/cr0s/warpdrive/world/WorldGenStructure.java

```diff
@@ -336,17 +350,34 @@ public void deployShip(final World world, final JumpShip jumpShip, final int tar
 					WarpDrive.logger.info(String.format("At index %d, skipping anchor block %s", index, jumpBlock.block));
 				}
 			} else {
-				index++;
-				if (WarpDrive.isDev && WarpDriveConfig.LOGGING_WORLD_GENERATION) {
+				//index++;	`//why is this here?
+				if (WarpDrive.isDev && LOGGING_WORLD_GENERATION) {
 					WarpDrive.logger.info(String.format("At index %d, deploying %s ",
 					                                    index, jumpBlock));
 				}
 				final BlockPos targetLocation = transformation.apply(jumpBlock.x, jumpBlock.y, jumpBlock.z);
 				final Block blockAtTarget = world.getBlockState(targetLocation).getBlock();
 				if (blockAtTarget == Blocks.AIR || Dictionary.BLOCKS_EXPANDABLE.contains(blockAtTarget)) {
 					jumpBlock.deploy(null, world, transformation);
+					
+					//if insertion is specified, do check
+					if(insertions != null){
+						for(Insertion insertion : insertions){
+							if(insertion.isMatching(jumpBlock)){
+								//calculate where the block went
+								BlockPos deployedLocation = transformation.apply(jumpBlock.x, jumpBlock.y, jumpBlock.z);
+								
+								//generate actual loot
+								fillInventoryWithLoot(world, rand, deployedLocation,
+								insertion, insertion.getMinQuantity(),
+								(int)Math.floor(insertion.getMaxQuantity()/2.0), (int)Math.ceil(insertion.getMaxQuantity()/2.0),
```

Comment:
Consider passing the actual value and 0 instead of some overcomplicated rounding after division?  
`insertion.getMaxQuantity(), 0,`